src/terminal/adapter/terminalOutput.cpp

```diff
@@ -17,104 +17,89 @@ TerminalOutput::TerminalOutput() noexcept
     _gsetTranslationTables.at(3) = Latin1;
 }
 
-bool TerminalOutput::Designate94Charset(size_t gsetNumber, const std::pair<wchar_t, wchar_t> charset)
+bool TerminalOutput::Designate94Charset(size_t gsetNumber, const VTID charset)
 {
-    switch (charset.first)
+    switch (charset)
     {
-    case L'B': // US ASCII
-    case L'1': // Alternate Character ROM
+    case VTID("B"): // US ASCII
+    case VTID("1"): // Alternate Character ROM
         return _SetTranslationTable(gsetNumber, Ascii);
-    case L'0': // DEC Special Graphics
-    case L'2': // Alternate Character ROM Special Graphics
+    case VTID("0"): // DEC Special Graphics
+    case VTID("2"): // Alternate Character ROM Special Graphics

```

Comment:
i wonder if there's a lookup table optimization we could eventually throw together? 

```
std::make_array(...
pair{VTID("<"), DecSupplemental},
...
)
```

with a `lower_bound` search like we used in CodepointWidthDetector?

(not a nit, or feedback for this review specifically :smile:)

Reply:
(or perhaps a library like `frozen` or some other hand-rolled constexpr map fixture)

Reply:
Hmmm.... that would assumedly need the array to be sorted on the VTID, which isn't a very obvious order, and is the kind of think you could easily get wrong. You'd also lose the readability of the logical order (e.g. grouping the fallback id with it's primary id), which is a bit of negative. The switch does feel unnecessary repetitive now that it's a flat list, but I'd probably still prefer it to an array map if that array had to be sorted. In terms of efficiency I wouldn't have thought there'd much difference - I think the compiler is quite good at optimising a switch.

Reply:
Yeah, I wouldn't be averse to a regular map -- it's static and relatively small, so the one-time hit of constructing it would be minimal.

I recently learned that VS 2019 16.6 (which is _ever so slightly_ too new for us to rely on to be ubiquitous) has constexpr `std::sort`. I was thinking about this because I'd been reading an e-mail from Stephen T. Lavavej (who incidentally works on our STL here at microsoft) about constexpr sorted maps and how they're now fully possible :smile:

I can think of a couple places we'd want to use one, but I'll just have to wait until we are sure folks have moved off <=16.5 😁

Reply:
I was just wondering whether you could get a constexpr sort - that would be really useful!

Reply:
#7323 :smile:

Reply:
How do the switch statements now look in machine code?

It seems to me that a C++ compiler could transform a lengthy uint64_t switch to a perfect hash that could then be faster than a binary search.

Reply:
They tend to compile to a mix and match of things. So when there's a big gap between items you'll get a conditional jump splitting them up, but if there's a bunch in sequence then that can be handled with a jump table.

Reply:
Yeah -- it's likely better to let the compiler handle this one, but it's always good to have more tools in our belts!